src/main/java/org/cascadebot/cascadebot/data/objects/donation/Tier.java

```diff
@@ -0,0 +1,239 @@
+    }
+
+    public void addFlag(Flag flag) {
+        flags.add(flag);
+    }
+
+    public Tier getParent() {
+        if (parent == null) return null;
+        return tiers.get(parent);
+    }
+
+    public Set<Flag> getFlags() {
+        Set<Flag> flags = new HashSet<>(this.flags);
+        if (getParent() != null) {
+            flags.addAll(getParent().getFlags());
+        }
+        return flags;
+    }
+
+    public Flag getFlag(String id) {
+        Flag return_flag = flags.stream().filter(flag -> flag.getId().equals(id)).findFirst().orElse(null);
+        if (parent != null && return_flag == null) {
+            return_flag = tiers.get(parent).getFlag(id);
+        }
+        return return_flag;
+    }
+
+    public boolean isTierParent(String tier) {
+        if (this.parent == null) {
+            return false;
+        }
+        if (this.parent.equals(tier)) {
+            return true;
+        }
+        if (this.parent.isEmpty()) {
+            return false;
+        }
+        return tiers.get(parent).isTierParent(tier);
+    }
+
+    /**
+     * Returns the benefits gave in this tier to a guild.
+     *
+     * @param locale The locale to use for translations
+     * @return The guild benefits gave in this tier
+     */
+    public String getGuildTierString(Locale locale, List<String> idsUsed) {
+        if (idsUsed == null) {
+            idsUsed = new ArrayList<>();
+        }
+        StringBuilder tierStringBuilder = new StringBuilder();
+        for (Flag flag : flags) {
+            if (!idsUsed.contains(flag.getId())) {
+                if (flag.scope.equals(Flag.FlagScope.GUILD)) {
+                    idsUsed.add(flag.getId());
+                    tierStringBuilder.append(" - **").append(flag.getName(locale)).append(":** ").append(flag.getDescription(locale)).append('\n');
+                }
+            }
+        }
+        for (TierExtra extra : extras) {
+            if (extra.scope.equals(Flag.FlagScope.GUILD)) { //Ignore t-shirt related extra stuff as that isn't a guild thing.
+                String extraStr = Language.getLanguage(locale).getString(extra.path).orElse("No language string defined");
+                tierStringBuilder.append(" - ").append(extraStr).append('\n');
+            }
+        }
+
+        if (parent != null && !parent.equals("default")) {
+            tierStringBuilder.append('\n');
+            tierStringBuilder.append("**__Inherited from ").append(parent).append(":__**\n");
+            tierStringBuilder.append(tiers.get(parent).getGuildTierString(locale, idsUsed));
+        }
+
+        return tierStringBuilder.toString();
+    }
+
+    /*
+     * Get's this tiers text as it would appear on patreon, and other front end stuffs.
+     *
+     * @param locale he locale to use for translations
+     * @return This tiers text as it would appear on patreon, and other front end stuffs.
+     */
+    /*
+    public String toDonateString(Locale locale) {
+        return "";
+    }
```

Comment:
Is this possible to be done in this PR?

Reply:
I asked about this the other day how we would display it front end and got not much of a response so

Reply:
**Waiting for meeting**